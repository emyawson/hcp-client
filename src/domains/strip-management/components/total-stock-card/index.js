export * from './total-stock-card.component';
