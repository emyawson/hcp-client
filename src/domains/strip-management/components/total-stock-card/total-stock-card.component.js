import React from 'react';

import { StripStockIcon } from 'src/assets/icons';
import { Badge, Caption, Card, LocalizedText } from 'src/components';

import {
  TotalStockCardLineItemDiv,
  TotalStockCardLineItemRightDiv,
  TotalStockSmallCardSeparatorDiv,
  CenteredFlexRowDiv,
  CenteredFlexRowWithMarginDiv,
  TotalStockContentWrapperDiv,
  TotalStockSubheadlinesDiv,
  TotalStockHeadline,
  QuantityDisplayDiv,
} from './total-stock-card.style';

export const TotalStockCard = ({ aggregateStock }) => {
  const { stripStock, tubeStock, lastShipmentDate } = aggregateStock;
  return (
    <Card cardStyles={['secondary']}>
      <TotalStockContentWrapperDiv>
        <TotalStockHeadline>
          <LocalizedText textKey="orgStock.totalStock.title" />
        </TotalStockHeadline>

        <TotalStockSubheadlinesDiv>
          <Caption>
            <LocalizedText textKey="orgStock.totalStock.lastAddedDate" />
            <span>: {lastShipmentDate}</span>
          </Caption>
        </TotalStockSubheadlinesDiv>

        <CenteredFlexRowWithMarginDiv>
          <Badge size={120} icon={<StripStockIcon height={56} />} />
        </CenteredFlexRowWithMarginDiv>
        <CenteredFlexRowDiv>
          <TotalStockCardLineItemRightDiv>
            <QuantityDisplayDiv>{tubeStock}</QuantityDisplayDiv>
            <Caption>
              <LocalizedText textKey="orgStock.totalStock.tubes" />
            </Caption>
          </TotalStockCardLineItemRightDiv>
          <TotalStockSmallCardSeparatorDiv />
          <TotalStockCardLineItemDiv>
            <QuantityDisplayDiv>{stripStock}</QuantityDisplayDiv>
            <Caption>
              <LocalizedText textKey="orgStock.totalStock.strips" />
            </Caption>
          </TotalStockCardLineItemDiv>
        </CenteredFlexRowDiv>
      </TotalStockContentWrapperDiv>
    </Card>
  );
};
