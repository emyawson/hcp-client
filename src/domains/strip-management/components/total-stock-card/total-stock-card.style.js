import styled from 'styled-components';

import { spacing, fontSize } from 'src/core';
import { Headline } from 'src/components';
import { combineRems } from 'src/utils';

const totalStockContentRhythm = combineRems(spacing.three, spacing.four);
export const CenteredFlexRowDiv = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
`;
export const CenteredFlexRowWithMarginDiv = CenteredFlexRowDiv.extend`
  margin-bottom: ${totalStockContentRhythm};
`;

export const TotalStockContentWrapperDiv = styled.div`
  padding: ${spacing.three} 0;
  text-align: center;
`;

export const TotalStockSubheadlinesDiv = styled.div`
  margin-bottom: ${spacing.four};
`;

export const TotalStockHeadline = Headline.extend`
  margin-bottom: ${spacing.three};
`;

export const TotalStockCardLineItemDiv = styled.div`
  display: flex;
  flex-direction: column;
  text-align: left;
`;

export const TotalStockCardLineItemRightDiv = TotalStockCardLineItemDiv.extend`
  text-align: right;
`;

export const TotalStockSmallCardSeparatorDiv = styled.div`
  border-left: 1px solid ${props => props.theme.colors.grayLight};
  display: inline-flex;
  margin: 0 ${combineRems(spacing.two, spacing.three)};
`;

export const QuantityDisplayDiv = styled.div`
  font-size: ${fontSize.display2};
  line-height: 1;
  margin-bottom: ${spacing.two};
`;
