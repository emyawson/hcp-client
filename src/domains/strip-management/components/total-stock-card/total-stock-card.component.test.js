import React from 'react';
import { shallow } from 'enzyme';

import { TotalStockCard } from './total-stock-card.component';

describe('Total stock card component', () => {
  it('Renders correctly', () => {
    const mockProps = {
      aggregateStock: {
        stripStock: 100,
        tubeStock: 5000,
        lastShipmentDate: '2018-02-27T00:00:00.000Z',
      },
    };
    const wrapper = shallow(<TotalStockCard {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
