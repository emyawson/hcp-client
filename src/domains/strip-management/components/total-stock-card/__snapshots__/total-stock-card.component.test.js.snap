// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Total stock card component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    aggregateStock={
      Object {
        "lastShipmentDate": "2018-02-27T00:00:00.000Z",
        "stripStock": 100,
        "tubeStock": 5000,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "cardStyles": Array [
        "secondary",
      ],
      "children": <styled.div>
        <styled.h5>
          <LocalizedText
            textKey="orgStock.totalStock.title"
          />
        </styled.h5>
        <styled.div>
          <styled.p>
            <LocalizedText
              textKey="orgStock.totalStock.lastAddedDate"
            />
            <span>
              : 
              2018-02-27T00:00:00.000Z
            </span>
          </styled.p>
        </styled.div>
        <styled.div>
          <Badge
            icon={
              <Unknown
                height={56}
              />
            }
            size={120}
          />
        </styled.div>
        <styled.div>
          <styled.div>
            <styled.div>
              5000
            </styled.div>
            <styled.p>
              <LocalizedText
                textKey="orgStock.totalStock.tubes"
              />
            </styled.p>
          </styled.div>
          <styled.div />
          <styled.div>
            <styled.div>
              100
            </styled.div>
            <styled.p>
              <LocalizedText
                textKey="orgStock.totalStock.strips"
              />
            </styled.p>
          </styled.div>
        </styled.div>
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.h5>
            <LocalizedText
              textKey="orgStock.totalStock.title"
            />
          </styled.h5>,
          <styled.div>
            <styled.p>
              <LocalizedText
                textKey="orgStock.totalStock.lastAddedDate"
              />
              <span>
                : 
                2018-02-27T00:00:00.000Z
              </span>
            </styled.p>
          </styled.div>,
          <styled.div>
            <Badge
              icon={
                <Unknown
                  height={56}
                />
              }
              size={120}
            />
          </styled.div>,
          <styled.div>
            <styled.div>
              <styled.div>
                5000
              </styled.div>
              <styled.p>
                <LocalizedText
                  textKey="orgStock.totalStock.tubes"
                />
              </styled.p>
            </styled.div>
            <styled.div />
            <styled.div>
              <styled.div>
                100
              </styled.div>
              <styled.p>
                <LocalizedText
                  textKey="orgStock.totalStock.strips"
                />
              </styled.p>
            </styled.div>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <LocalizedText
              textKey="orgStock.totalStock.title"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "textKey": "orgStock.totalStock.title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.p>
              <LocalizedText
                textKey="orgStock.totalStock.lastAddedDate"
              />
              <span>
                : 
                2018-02-27T00:00:00.000Z
              </span>
            </styled.p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <LocalizedText
                  textKey="orgStock.totalStock.lastAddedDate"
                />,
                <span>
                  : 
                  2018-02-27T00:00:00.000Z
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "textKey": "orgStock.totalStock.lastAddedDate",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    ": ",
                    "2018-02-27T00:00:00.000Z",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  ": ",
                  "2018-02-27T00:00:00.000Z",
                ],
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Badge
              icon={
                <Unknown
                  height={56}
                />
              }
              size={120}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "icon": <Unknown
                height={56}
              />,
              "size": 120,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.div>
                  5000
                </styled.div>
                <styled.p>
                  <LocalizedText
                    textKey="orgStock.totalStock.tubes"
                  />
                </styled.p>
              </styled.div>,
              <styled.div />,
              <styled.div>
                <styled.div>
                  100
                </styled.div>
                <styled.p>
                  <LocalizedText
                    textKey="orgStock.totalStock.strips"
                  />
                </styled.p>
              </styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div>
                    5000
                  </styled.div>,
                  <styled.p>
                    <LocalizedText
                      textKey="orgStock.totalStock.tubes"
                    />
                  </styled.p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": 5000,
                  },
                  "ref": null,
                  "rendered": 5000,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <LocalizedText
                      textKey="orgStock.totalStock.tubes"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "orgStock.totalStock.tubes",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div>
                    100
                  </styled.div>,
                  <styled.p>
                    <LocalizedText
                      textKey="orgStock.totalStock.strips"
                    />
                  </styled.p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": 100,
                  },
                  "ref": null,
                  "rendered": 100,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <LocalizedText
                      textKey="orgStock.totalStock.strips"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "orgStock.totalStock.strips",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "cardStyles": Array [
          "secondary",
        ],
        "children": <styled.div>
          <styled.h5>
            <LocalizedText
              textKey="orgStock.totalStock.title"
            />
          </styled.h5>
          <styled.div>
            <styled.p>
              <LocalizedText
                textKey="orgStock.totalStock.lastAddedDate"
              />
              <span>
                : 
                2018-02-27T00:00:00.000Z
              </span>
            </styled.p>
          </styled.div>
          <styled.div>
            <Badge
              icon={
                <Unknown
                  height={56}
                />
              }
              size={120}
            />
          </styled.div>
          <styled.div>
            <styled.div>
              <styled.div>
                5000
              </styled.div>
              <styled.p>
                <LocalizedText
                  textKey="orgStock.totalStock.tubes"
                />
              </styled.p>
            </styled.div>
            <styled.div />
            <styled.div>
              <styled.div>
                100
              </styled.div>
              <styled.p>
                <LocalizedText
                  textKey="orgStock.totalStock.strips"
                />
              </styled.p>
            </styled.div>
          </styled.div>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.h5>
              <LocalizedText
                textKey="orgStock.totalStock.title"
              />
            </styled.h5>,
            <styled.div>
              <styled.p>
                <LocalizedText
                  textKey="orgStock.totalStock.lastAddedDate"
                />
                <span>
                  : 
                  2018-02-27T00:00:00.000Z
                </span>
              </styled.p>
            </styled.div>,
            <styled.div>
              <Badge
                icon={
                  <Unknown
                    height={56}
                  />
                }
                size={120}
              />
            </styled.div>,
            <styled.div>
              <styled.div>
                <styled.div>
                  5000
                </styled.div>
                <styled.p>
                  <LocalizedText
                    textKey="orgStock.totalStock.tubes"
                  />
                </styled.p>
              </styled.div>
              <styled.div />
              <styled.div>
                <styled.div>
                  100
                </styled.div>
                <styled.p>
                  <LocalizedText
                    textKey="orgStock.totalStock.strips"
                  />
                </styled.p>
              </styled.div>
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <LocalizedText
                textKey="orgStock.totalStock.title"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "textKey": "orgStock.totalStock.title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <styled.p>
                <LocalizedText
                  textKey="orgStock.totalStock.lastAddedDate"
                />
                <span>
                  : 
                  2018-02-27T00:00:00.000Z
                </span>
              </styled.p>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <LocalizedText
                    textKey="orgStock.totalStock.lastAddedDate"
                  />,
                  <span>
                    : 
                    2018-02-27T00:00:00.000Z
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "textKey": "orgStock.totalStock.lastAddedDate",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      ": ",
                      "2018-02-27T00:00:00.000Z",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    ": ",
                    "2018-02-27T00:00:00.000Z",
                  ],
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <Badge
                icon={
                  <Unknown
                    height={56}
                  />
                }
                size={120}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "icon": <Unknown
                  height={56}
                />,
                "size": 120,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.div>
                  <styled.div>
                    5000
                  </styled.div>
                  <styled.p>
                    <LocalizedText
                      textKey="orgStock.totalStock.tubes"
                    />
                  </styled.p>
                </styled.div>,
                <styled.div />,
                <styled.div>
                  <styled.div>
                    100
                  </styled.div>
                  <styled.p>
                    <LocalizedText
                      textKey="orgStock.totalStock.strips"
                    />
                  </styled.p>
                </styled.div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.div>
                      5000
                    </styled.div>,
                    <styled.p>
                      <LocalizedText
                        textKey="orgStock.totalStock.tubes"
                      />
                    </styled.p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": 5000,
                    },
                    "ref": null,
                    "rendered": 5000,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": <LocalizedText
                        textKey="orgStock.totalStock.tubes"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "textKey": "orgStock.totalStock.tubes",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.div>
                      100
                    </styled.div>,
                    <styled.p>
                      <LocalizedText
                        textKey="orgStock.totalStock.strips"
                      />
                    </styled.p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": 100,
                    },
                    "ref": null,
                    "rendered": 100,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": <LocalizedText
                        textKey="orgStock.totalStock.strips"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "textKey": "orgStock.totalStock.strips",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
